examples/threading/threading.c

```diff
@@ -0,0 +1,96 @@
+#include "threading.h"
+#include <pthread.h>
+#include <unistd.h>
+#include <time.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+#define MS_PER_SEC  (1000)
+#define NS_PER_MS   (1000000)
+
+struct timespec ms_to_timespec(uint32_t ms){
+    const uint32_t sec = ms / MS_PER_SEC;
+    ms %= MS_PER_SEC;
+    return (struct timespec) {.tv_sec = sec, .tv_nsec = ms * NS_PER_MS};
+}
+
+int sleep_ms(uint32_t ms){
+    struct timespec waiting_time = ms_to_timespec(ms);
+    struct timespec remaining_time;
+    while(nanosleep(&waiting_time, &remaining_time) == -1){
+        if(errno == EINTR){
+            waiting_time = remaining_time;
+        }else{
+            return EXIT_FAILURE;
+        }
+    }
+    return EXIT_SUCCESS;
+}
+
+void* threadfunc(void* thread_param)
+{
+
+    // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    struct thread_data* thread_func_args = (struct thread_data*) thread_param;
+    // Assume success, any test can overwrite with failure
+    thread_func_args->thread_complete_success = true;
```

Comment:
@bennowotny has the thread completed with success at this point?  Remember that `thread_param` is shared outside this module.